### PR TITLE
Update subworld handling of collisions and animation

### PR DESCRIPTION
**Pull request by TikiBear**
Update subworld handling of collisions and animation.

### CODE REVIEW

**Comment by TikiBear** [comment]
This is what I did to make subworlds work in The Flying Cauldron and Duck Knee.

File: korman/exporter/physics.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def generate_physical(self, bo, so, **kwargs):
 
             # Got subworlds?
             subworld = bo.plasma_object.subworld
-            if self.is_dedicated_subworld(subworld, sanity_check=False):
+            if subworld is not None and self.is_dedicated_subworld(subworld, sanity_check=False):
```
Comment (Hoikas) [inline korman/exporter/physics.py:160]: It looks like `is_dedicated_subworld()` already returns `False` for subworlds that are `None`.

File: korman/exporter/physics.py [hunk 2]
```diff
@@ -212,11 +212,11 @@ def _iter_object_tree(bo, stop_at_subworld):
                 bo_xformed = bo.plasma_object.has_transform_animation
 
                 # MOUL: only objects that have animation data are kPhysAnim
-                if ver != pvMoul or bo_xformed:
+                if mod.enabled and (ver != pvMoul or bo_xformed):
```
Comment (Hoikas) [inline korman/exporter/physics.py:215]: This could break animated clickables and regions because their physics objects are exported by nodes, not the collision modifier.

File: korman/exporter/physics.py [hunk 3]
```diff
@@ -212,11 +212,11 @@ def _iter_object_tree(bo, stop_at_subworld):
...
                     _set_phys_prop(plSimulationInterface.kPhysAnim, simIface, physical)
                 # PotS: objects inheriting parent animation only are not pinned
                 # MOUL: animated objects in subworlds are not pinned
-                if bo_xformed and (ver != pvMoul or subworld is None):
+                if (bo_xformed and (ver != pvMoul or subworld is None)) or ((ver != pvMoul) and subworld is not None and (not bo_xformed)):
```
Comment (Hoikas) [inline korman/exporter/physics.py:219]: I suspect this is what's fixing subworlds, but the `kPinned` property only has an effect in PotS. The conditional is really hard to follow, so I'm not really able to glance in and see what's happening.

**Comment by Hoikas** [comment]
@julyfortoday Did some work on demonstrating the subworld problems on the forum. I wasn't really able to understand his sample blend though, and edits to his posts indicated that he had figured out a number of the problems. Maybe he'll chime in? *cross fingers*

**Comment by TikiBear** [comment]
It sounds as though my approach is too naive. I'm fine standing down. I can only test on PotS, anyway. With everyone's helpful input I can watch for unexpected side effects going forward.
